### PR TITLE
Add parquet_ignore_statistics session property to iceberg and delta

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description
Add parquet_ignore_statistics session property to iceberg and delta

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Iceberg, Delta Lake
* Add `parquet_ignore_statistics` catalog session property. ({issue}`20228`)
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetReaderConfig.java [hunk 1]
```diff
@@ -37,13 +37,11 @@ public class ParquetReaderConfig
 
     private ParquetReaderOptions options = new ParquetReaderOptions();
 
-    @Deprecated
```
Comment (findinpath) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetReaderConfig.java:40]: Was it intentional that the session property `parquet_ignore_statistics` was introduced only for Hive, but not for Iceberg (Delta Lake connector didn't exist at the time) in 1981cefb ?

cc @findepi

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSessionProperties.java [hunk 2]
```diff
@@ -448,6 +454,11 @@ public static DataSize getParquetSmallFileThreshold(ConnectorSession session)
         return session.getProperty(PARQUET_SMALL_FILE_THRESHOLD, DataSize.class);
     }
 
+    public static boolean isParquetIgnoreStatistics(ConnectorSession session)
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSessionProperties.java:457]: Please add for both Iceberg & Delta at least a test like the following to ensure that the new session property is effective

https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java#L5927-L5961

At best we'd have a table with data files containing broken stats registered from resources.

**Comment by findinpath** [comment]
LGTM overall, but the PR is missing corresponding tests for the new functionality.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetReaderConfig.java (hunk 1 above)
Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetReaderConfig.java:40]: it was originally added to Trino due to a bug in Trino which caused us to ingest corrupted statistics.

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@raunaqmorarka do you have examples of corrupted Parquet stats for Iceberg or Delta?
would it be possible to automatically skip them, instead of relying on a user/administrator to turn the config?

**Comment by raunaqmorarka** [comment]
> @raunaqmorarka do you have examples of corrupted Parquet stats for Iceberg or Delta? would it be possible to automatically skip them, instead of relying on a user/administrator to turn the config?

We have a scenario reported in https://github.com/trinodb/trino/pull/20212#issuecomment-1869964174 where the user through some other tools managed to create a parquet file in iceberg where the partitioned column stats say it is all nulls but that's not the case in actual data. There are now many libraries of varying quality for writing parquet files. The ability to ignore stats through session property is a useful debugging tool when tackling issues with wrong results.
We can skip stats automatically only we have some known pattern of incorrectness from a specific library/tool. We don't have enough details about how the file was produced at the moment to know whether it's a user mistake or a library bug.

**Pull request merged**